pinecone/grpc/vector_factory_grpc.py

```diff
@@ -0,0 +1,122 @@
+import numbers
+
+from collections.abc import Iterable, Mapping
+from typing import Union, Tuple, Dict
+
+from google.protobuf.struct_pb2 import Struct
+
+from .utils import dict_to_proto_struct
+from ..utils import fix_tuple_length, convert_to_list
+from ..utils.constants import REQUIRED_VECTOR_FIELDS, OPTIONAL_VECTOR_FIELDS
+from ..data import (
+    VectorDictionaryMissingKeysError, 
+    VectorDictionaryExcessKeysError, 
+    VectorTupleLengthError, 
+    SparseValuesTypeError, 
+    SparseValuesMissingKeysError, 
+    SparseValuesDictionaryExpectedError, 
+    MetadataDictionaryExpectedError 
+)
+
+from pinecone.core.grpc.protos.vector_service_pb2 import (
+    Vector as GRPCVector,
+    QueryVector as GRPCQueryVector,
+    SparseValues as GRPCSparseValues,
+)
+from pinecone import (
+    Vector as NonGRPCVector, 
+    SparseValues as NonGRPCSparseValues
+)
+
+class VectorFactoryGRPC:
+    @staticmethod
+    def build(item: Union[GRPCVector, NonGRPCVector, Tuple, Dict]) -> GRPCVector:
+        if isinstance(item, GRPCVector):
+            return item
+        elif isinstance(item, NonGRPCVector):
```

Comment:
I don't think it would, but would the `VectorFactory` benefit from having the flexibility of `GRPCVector` being added to the union in `build`?

Reply:
Perhaps in some circumstances, but it's probably a lot less likely to happen than people trying to pass the `NonGRPCVector` object into into the GRPC-flavored client. The GRPC-flavored version of this object is annoying because you have to convert metadata dicts yourself into `Struct` before passing the kwarg param. So the ergonomics leave a lot to be desired.